var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -80,7 +81,11 @@ class NetcdfC(AutotoolsPackage):
     variant("zstd", default=True, description="Enable ZStandard compression", when="@4.9.0:")
     variant("optimize", default=True, description="Enable -O2 for a more optimized lib")
     variant("nczarr", default=True, description="Enable zarr storage support", when="@4.8.0:")
-    variant("byterange", default=False, description="Allow byte-range I/O")
+    # New byte-range I/O option doesn't compile with 4.9.1, will be fixed in 4.9.2
+    # https://github.com/Unidata/netcdf-c/issues/2614
+    variant("byterange", default=False, description="Allow byte-range I/O", when="@4.9.1")
+    variant("byterange", default=True, description="Allow byte-range I/O", when="@4.9.2:")
```

Comment:
As long as we're at it, how about adding `variant("byterange", default=False, description="Allow byte-range I/O", when="@4.7.0:4.9.0")`? In any case I would lean toward setting it default=False because it's off by default in the build system itself (both configure and cmake).

Reply:
Yes, I agree. I didn't know that it was available in 4.7.0 already. I'll make the change as follows (not that it goes up to 4.9.1):
```
    variant("byterange", default=False, description="Allow byte-range I/O", when="@4.7.0:4.9.1")
    variant("byterange", default=True, description="Allow byte-range I/O", when="@4.9.2:")
```

Reply:
Very annoying, but for some reason the above suggestion makes `+byterange` the default, even for `4.9.0`. Looks like you convinced me, for now I'll just use
```
variant("byterange", default=False, description="Allow byte-range I/O", when="@4.7.0:")
```
and we deal with enabling it for newer versions later.